src/object/cnode.c

```diff
@@ -806,6 +806,21 @@ bool_t PURE isMDBParentOf(cte_t *cte_a, cte_t *cte_b)
         break;
     }
 
+#ifdef CONFIG_ALLOW_SMC_CALLS
```

Comment:
This would benefit from some arch-specific functions to call out to, but they don't currently exist... Thoughts?

Reply:
I suppose it could also be wrapped in an architecture ifdef so that if we ever make an architecture specific `cnode.c` file, then it is obvious that chunk of code should move.